src/RVRP_data.jl

```diff
@@ -1,118 +1,99 @@
     id::String # optional
+    index::int
     coord::Coord # optional
-    index::Int # optional
 end
 
 struct TimeWindow
     tw_start::Float64
     tw_end::Float64
 end
 
-struct Costs
-    fixed::Float64
-    distance::Float64
-    time::Float64
-    service::Float64
-    wait::Float64
-end
-
-struct VehicleType
-    id::String
-    capacity::Int
-    costs::Costs
-end
-
 struct Depot <: AbstractNode
     location::Location
     time_windows::Vector{TimeWindow} # optional
 end
 
-struct Pickup <: AbstractNode
-    location::Location
-    capacity_demand::Float64
-    time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional      
-    req_id::String
-end
-
-struct Delivery <: AbstractNode
-    location::Location
-    capacity_demand::Float64
-    time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional
-    req_id::String
-end
-
 struct Operation <: AbstractNode
-    start_location::Location
-    end_location::Location
-    required_capacity::Float64
-    delta_capacity::Float64
+    location::Location
+    resource_consumptions::Dict{RESOURCE,Float64}
     time_windows::Vector{TimeWindow} # optional
-    duration::Float64 # optional      
-    req_id::String
+    req_index::Int
+    operation_type::OPERATION # Pickup, Delivery, Cleaning, ...
 end
 
-abstract type AbstractRequest end
-
-struct PickupRequest <: AbstractRequest
-    id::String
-    node::Pickup
-end    
-
-struct DeliveryRequest <: AbstractRequest
+struct SimpleRequest
     id::String
-    node::Delivery
-end    
-
-struct OperationRequest <: AbstractRequest
-    id::String
-    node::Operation
-end    
+    index::Int
+    operation::Operation # Single Requests have only one operation
+    request_type::REQUEST # SingleOperation, Shipment, ...
+end
 
-struct Shipment <: AbstractRequest
+struct ComplexRequest
     id::String
-    pickup::Pickup
-    delivery::Delivery
+    index::Int
+    operations::Vector{Operation} # Sequence of operations ; can be limited to one; can be a pair of Pickup and Delivery, or a triplets including a cleaning first, etc
+    request_type::REQUEST # SingleOperation, Shipment, ...
 end
 
-abstract type AbstractVehicle end
+struct VehicleBaseType
+    fixed_cost::Float64
+    resource_unit_prices::Vector{Float64}
+    resource_capacities::Vector{Float64}
+end
 
-struct Vehicle <: AbstractVehicle
+struct VehicleType
     id::String
+    index::Int
     depot::Depot # optional
-    v_type::VehicleType
+    base_type::VehicleType
     time_schedule::TimeWindow
     return_to_depot::Bool
     infinite_copies::Bool
-    initial_load::Float64
-    picked_shipments::Vector{Shipment}
+    nb_of_copies::Int
+    resource_intial_states::Vector{Float64}
+    ongoing_requests::Vector{Pair{Request,Int}}
 end
 
 struct RvrpProblem
     problem_id::String
     problem_type::ProblemType
-    vehicles::Vector{Vehicle}
-    vehicle_types::Vector{VehicleType}
+    vehicles::Vector{VehicleType}
```

Comment:
We should also add the set of Resource. Only these would be used the as keys in all Resource dicts

Reply:
Would it be a Set{RESOURCE} ?

Reply:
I prefer a Vector{RESOURCE} but knowing the size, it wouldn't matter much for julia performance. What is the equivalent of Set in Json?